sbnobj/Common/Calibration/TrackCaloSkimmerObj.h

```diff
@@ -0,0 +1,156 @@
+#ifndef SBN_TrackCaloSkimmerObj
+#define SBN_TrackCaloSkimmerObj
+
+#include <vector>
+
+namespace sbn {
+  struct HitInfo {
+    float integral;
+    float sumadc;
+    float width;
+    float pitch;
+    float dqdx;
+    float rr;
+    float x;
+    float y;
+    float z;
+    float time;
+    uint16_t wire;
+    uint16_t plane;
+    uint16_t tpc;
+    uint16_t mult;
+    uint16_t i_snippet;
+    int16_t start;
+    int16_t end;
+    bool ontraj;
+    bool oncalo;
+
+    // raw adc info
+    short tdc0;
+    std::vector<short> adcs;
+
+    HitInfo():
+      integral(-1),
+      sumadc(-1),
+      width(-1),
+      pitch(-1),
+      dqdx(-1),
+      rr(-1),
+      x(-1),
+      y(-1),
+      z(-1),
+      time(-1),
+      wire((uint16_t)-1),
+      plane((uint16_t)-1),
+      tpc((uint16_t)-1),
+      mult((uint16_t)-1),
+      i_snippet((uint16_t)-1),
+      start(-1),
+      end(-1),
+      ontraj(false),
+      oncalo(false),
+      tdc0(-1) {}
+  };
+
+  struct MetaInfo {
+    int run;
+    int evt;
+    int subrun;
+    int ifile;
+    int iproc;
+
+   MetaInfo():
+     run(-1),
+     evt(-1),
+     subrun(-1),
+     ifile(-1),
+     iproc(-1) {}
+  };
+
+  struct TrackInfo {
+    MetaInfo meta;
+    std::vector<HitInfo> hits0; 
```

Comment:
Is there a reason to include the hitinfo in the TrackInfo object like this? Rather than do an association (in art)? Or is this for ease/convenience of setting up the output trees? I think we'd generally prefer to do a linkage by association rather than inclusion like this.

Reply:
This structure isn't really intended for use in art. It's just supposed to represent a relatively flat structure that can be encoded as an ntuple. So I don't think bringing in art things makes sense here.